configure.ac

```diff
@@ -251,6 +251,22 @@ then
 fi
 AC_MSG_RESULT($RPATH)
 
+dnl cygpath: Converts POSIX and Windows format paths
+dnl -m, --mixed    like --windows, but with regular slashes (C:/WINNT)
+dnl -u             (default) print Unix form of NAMEs (/cygdrive/c/winnt)
+dnl -w             print Windows form of NAMEs (C:\WINNT)
+dnl See: https://cygwin.com/cygwin-ug-net/cygpath.html
+AC_CHECK_TOOL([CYGPATH], [cygpath])
+dnl Fall back to convertpath
+AS_IF([test -z "$CYGPATH"],
+      [AC_MSG_CHECKING([alternative for cygpath])
+       AS_VAR_SET([CYGPATH], [`(cd ${srcdir} && pwd)`/Tools/convertpath])
+       AC_SUBST(CYGPATH)
+       AC_MSG_RESULT($CYGPATH)])
+      dnl '--mixed' and '-u' mimic cygpath '-m'
+      dnl '-w' mimic cygpath '-w'
+      dnl '-m' is bizzare. FIXME: why do we have this mode?
+
```

Comment:
Can we try remove this and go back to only using Tools/convertpath? That is more consistent as cygpath is not always available and in my experience over many years, keeps changing. As far as I know Tools/convertpath does all we need, it is a very simple implementation of cygpath.

Reply:
1. MSYS2 uses the tool `cygpath`, the current assumption that the tool exist with Cygwin only is incorrect. 
   I think it is always better to assume less and probe more.
   I try to follow the autoconf approach.
2. MSYS2 as Cygwin uses different mounting points
    for example `/mingw64/bin` Windows path is `C:\msys64\mingw64`
    The `convertpath`  convert solids to back slash, but do not handle mounting points.
3. Yes, `cygpath` may not be available, therefor we  `Fall back to convertpath`.



I did thought we can  probe in a windows environment.
By doing it like:

````
case $host in
  *-*-cygwin* | *-*-mingw* | *-*-msys*)
    AC_CHECK_TOOL([CYGPATH], [cygpath])
    ;;
esac
dnl Fall back to convertpath
AS_IF([test -z "$CYGPATH"],
...
````

Reply:
Hi @wsfulton,

I tested the fallback of using the `Tools/convertpath`.
I stand with my opinion that using `cygpath` when it is available is better than using the `Tools/convertpath`.
If you wish to simplify, there are many other places we can.
I have some suggestion, and I hope you have more.
However with the  `cygpath` tool I only see benefits over using the `Tools/convertpath`.